apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -806,42 +736,35 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     value
   end
 
-  def retrieve_endpoint_values(client, device_id, interface_descriptor, mapping) do
-    query_statement =
-      prepare_get_property_statement(
-        mapping.value_type,
-        false,
-        interface_descriptor.storage,
-        interface_descriptor.storage_type
-      )
+  def retrieve_property_values(realm, device_id, interface_descriptor, mapping) do
+    %InterfaceDescriptor{
+      storage_type: :multi_interface_individual_properties_dbtable,
+      interface_id: interface_id
+    } = interface_descriptor
 
-    query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(query_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:interface_id, interface_descriptor.interface_id)
-      |> DatabaseQuery.put(:endpoint_id, mapping.endpoint_id)
+    %Mapping{endpoint_id: endpoint_id, value_type: value_type} = mapping
+    column_name = astarte_type_to_db_column_name(value_type)
 
-    DatabaseQuery.call!(client, query)
-  end
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
-  defp prepare_get_property_statement(
-         value_type,
-         metadata,
-         table_name,
-         :multi_interface_individual_properties_dbtable
-       ) do
-    metadata_column =
-      if metadata do
-        ",metadata"
-      else
-        ""
-      end
+    all_values =
+      IndividualProperty
+      |> where(device_id: ^device_id, interface_id: ^interface_id, endpoint_id: ^endpoint_id)
+      |> put_query_prefix(keyspace_name)
+
+    Enum.map(all_values, fn %{:path => path, ^column_name => value} ->
+      # TODO The `metadata` field was ignored before (hardcoded). Should we keep ignoring it?
+      [{:path, path}, {column_name, value}]
+    end)
+  end
 
-    # TODO: should we filter on path for performance reason?
-    # TODO: probably we should sanitize also table_name: right now it is stored on database
-    "SELECT path, #{Astarte.Core.CQLUtils.type_to_db_column_name(value_type)} #{metadata_column} FROM #{table_name}" <>
-      " WHERE device_id=:device_id AND interface_id=:interface_id AND endpoint_id=:endpoint_id;"
+  defp astarte_type_to_db_column_name(astarte_type) do
+    astarte_type
+    |> Atom.to_string()
+    |> (fn t -> t <> "_value" end).()
+    # Those atoms exist in schemas
+    |> String.to_existing_atom()
```

Comment:
Why not keep using `Astarte.Core.CQLUtils.type_to_db_column_name`?
If we want to avoid depending on Astarte.Core for this, we can evaluate defining the same function on the appropriate Ecto schema's module.

Reply:
After IRL discussion, this was change in another way: move the `Enum.map` on the caller and leave the Queries module to just interact with the database